src/System.Text.Json/src/System/Text/Json/CustomUncheckedBitArray.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json
+{
+    // Inspired by BitArray
+    // https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/BitArray.cs
+    internal sealed class CustomUncheckedBitArray
+    {
+        private int[] _array;
+
+        public int Length { get; private set; }
```

Comment:
I would use explicit `_length` field here. It makes the code easier to understand and it makes the code cleaner.

Reply:
Actually, do you really need `Length` for what you are doing? It should not be necessary.

Reply:
It doesn't need to be public, but I need some way of knowing if the backing array is large enough or if I need to grow it.
```C#
if (index >= Length)
{
   Grow (index);
}
```

I could try and use `_array.Length` for that instead. That might actually work.